py/fiberassign/assign.py

```diff
@@ -823,57 +823,57 @@ def merge_results_tile_initialize(tgbufs, tgdtypes, tgshapes, skybufs,
 # AR we also propagate any *_TARGET existing column
 minimal_target_columns = OrderedDict([
     ('RELEASE', '>i2'),
-    ('BRICKNAME', 'S8'), 
-    ('BRICKID', '>i4') , 
-    ('BRICK_OBJID', '>i4'),                    
-    ('MORPHTYPE', 'S4'), 
-    ('RA', '>f8'), 
-    ('DEC', '>f8'), 
-    ('EBV', '>f4'), 
-    ('FLUX_G', '>f4'), 
-    ('FLUX_R', '>f4'), 
-    ('FLUX_Z', '>f4'), 
+    ('BRICKNAME', 'S8'),
+    ('BRICKID', '>i4') ,
+    ('BRICK_OBJID', '>i4'),
+    ('MORPHTYPE', 'S4'),
+    ('RA', '>f8'),
+    ('DEC', '>f8'),
+    ('EBV', '>f4'),
+    ('FLUX_G', '>f4'),
+    ('FLUX_R', '>f4'),
+    ('FLUX_Z', '>f4'),
     ('FLUX_W1', '>f4'), # AR for QA plots
     ('FLUX_W2', '>f4'), # AR for QA plots
-    ('FLUX_IVAR_G', '>f4'), 
-    ('FLUX_IVAR_R', '>f4'), 
+    ('FLUX_IVAR_G', '>f4'),
+    ('FLUX_IVAR_R', '>f4'),
     ('FLUX_IVAR_Z', '>f4'),
     ('FLUX_IVAR_W1', '>f4'), # https://github.com/desihub/fiberassign/issues/300
     ('FLUX_IVAR_W2', '>f4'), # https://github.com/desihub/fiberassign/issues/300
-    ('FIBERFLUX_G', '>f4'), 
-    ('FIBERFLUX_R', '>f4'), 
-    ('FIBERFLUX_Z', '>f4'), 
-    ('FIBERTOTFLUX_G', '>f4'), 
-    ('FIBERTOTFLUX_R', '>f4'), 
-    ('FIBERTOTFLUX_Z', '>f4'), 
-    ('REF_EPOCH', '>f4'), 
-    ('MASKBITS', '>i2'), 
-    ('SERSIC', '>f4'), 
-    ('SHAPE_R', '>f4'), 
-    ('SHAPE_E1', '>f4'), 
-    ('SHAPE_E2', '>f4'), 
-    ('REF_ID', '>i8'), 
-    ('REF_CAT', 'S2'), 
-    ('GAIA_PHOT_G_MEAN_MAG', '>f4'),  
-    ('GAIA_PHOT_BP_MEAN_MAG', '>f4'),  
-    ('GAIA_PHOT_RP_MEAN_MAG', '>f4'), 
-    ('PARALLAX', '>f4'),  
-    ('PMRA', '>f4'), 
-    ('PMDEC', '>f4'), 
-    ('PHOTSYS', 'S1'), 
-    ('TARGETID', '>i8'), 
-    #('DESI_TARGET', '>i8'), 
-    #('BGS_TARGET', '>i8'), 
+    ('FIBERFLUX_G', '>f4'),
+    ('FIBERFLUX_R', '>f4'),
+    ('FIBERFLUX_Z', '>f4'),
+    ('FIBERTOTFLUX_G', '>f4'),
+    ('FIBERTOTFLUX_R', '>f4'),
+    ('FIBERTOTFLUX_Z', '>f4'),
+    ('REF_EPOCH', '>f4'),
+    ('MASKBITS', '>i2'),
+    ('SERSIC', '>f4'),
+    ('SHAPE_R', '>f4'),
+    ('SHAPE_E1', '>f4'),
+    ('SHAPE_E2', '>f4'),
+    ('REF_ID', '>i8'),
+    ('REF_CAT', 'S2'),
+    ('GAIA_PHOT_G_MEAN_MAG', '>f4'),
+    ('GAIA_PHOT_BP_MEAN_MAG', '>f4'),
+    ('GAIA_PHOT_RP_MEAN_MAG', '>f4'),
+    ('PARALLAX', '>f4'),
+    ('PMRA', '>f4'),
+    ('PMDEC', '>f4'),
+    ('PHOTSYS', 'S1'),
+    ('TARGETID', '>i8'),
+    #('DESI_TARGET', '>i8'),
+    #('BGS_TARGET', '>i8'),
     #('MWS_TARGET', '>i8'),
-    #('CMX_TARGET', '>i8'), 
+    #('CMX_TARGET', '>i8'),
     #('SV1_DESI_TARGET', '>i8'),
     #('SV1_BGS_TARGET', '>i8'),
     #('SV1_MWS_TARGET', '>i8'),
-    #('SCND_TARGET', '>i8'), 
-    ('SUBPRIORITY', '>f8'), 
-    ('OBSCONDITIONS', '>i8'), 
-    ('PRIORITY_INIT', '>i8'), 
-    ('NUMOBS_INIT', '>i8')   
+    #('SCND_TARGET', '>i8'),
+    ('SUBPRIORITY', '>f8'),
+    ('OBSCONDITIONS', '>i8'),
+    ('PRIORITY_INIT', '>i8'),
+    ('NUMOBS_INIT', '>i4')
```

Comment:
Although NUMOBS_INIT as i4 is a more appropriate type than the original i8, this is another data model change for fiberassign output that creates bookkeeping hassle for downstream code that does things like stack these tables into a catalog.  We're already dealing with other type and column changes so this doesn't have to be a blocking factor, but I wanted to highlight this in case others anticipate problems.

Reply:
Ok, will change to i8.  This column is used by some of the target QA.  When it was removed, in the changes to the file format, this broke that QA.  However, if we really don't want to propagate this column, then I can disable the unit tests that depend on it.  I guess I was the only one using that target QA :-)

Reply:
Please don't re-add a column that had been removed unless there has been group discussion that re-adding it is really necessary.  I thought you were just changing type from i8 to i4 which could be disruptive but ok, but adding and subtracting columns is a bigger deal that shouldn't be done lightly now that we are actively observing.

Reply:
Ok, I guess my point was that removing the column broke things, but this went undetected since unit tests were not being run.  Restoring this column fixes the unit tests.  If restoring this column breaks downstream code, then I guess we should disable the target QA code that needs it.  I can do that with a separate small PR if you like.

Reply:
Removing NUMOBS_INIT was part of a purposeful trimming down of the fiberassign format that was discussed by the entire group and jointly decided upon.  It was an error that we weren't running unit tests and didn't realize it, leading us to not realize that it broke qa-fiberassign, but the fix now is to update qa-fiberassign to not need NUMOBS_INIT rather than adding that column back.  If re-removing it requires you to comment out the qa-fiberassign test now so that you can merge this and focus on the other more critical fiberassign+desimodel updates, ok.

Reply:
Just to be clear, the NUMOBS_INIT column was used by a function (qa_targets) which is separate from the `qa-fiberassign` script.  I will self-merge a PR shortly that re-removes the NUMOBS_INIT column and comments out the unit test of the qa_targets() function.